### PR TITLE
HVT-4279 add support for cloudwatch observability on vault_cluster resource

### PR DESCRIPTION
**Pull request by bosouza**
### :hammer_and_wrench: Description

Add CloudWatch option for observability providers on HCP Vault clusters. See [HVT-4279](https://hashicorp.atlassian.net/browse/HVT-4279). I manually tested updating the audit-logs and metrics settings using the TF provider on cluster creation and cluster updates. Didn't add an acceptance test because the increase in the test duration didn't seem justified by such a small change.

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:
```
$ make testacc TESTARGS='-run=TestAccVaultClusterAWS'
==> Checking that code complies with gofmt requirements...
golangci-lint run --config ./golangci-config.yml 
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultClusterAWS -timeout 360m -parallel=10
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/clients    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/consul     (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/input      (cached) [no tests to run]
=== RUN   TestAccVaultClusterAWS
=== PAUSE TestAccVaultClusterAWS
=== CONT  TestAccVaultClusterAWS
--- PASS: TestAccVaultClusterAWS (3339.18s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   3339.188s

...
```

[HVT-4279]: https://hashicorp.atlassian.net/browse/HVT-4279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/provider/data_source_vault_cluster.go [hunk 1]
```diff
@@ -192,6 +212,26 @@ If a project is not configured in the HCP Provider config block, the oldest proj
 							Type:        schema.TypeString,
 							Computed:    true,
 						},
+						"cloudwatch_access_key_id": {
+							Description: "CloudWatch access key ID for streaming audit logs",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
```
Comment (mercedesbh) [inline internal/provider/data_source_vault_cluster.go:219]: Looks like we need `cloudwatch_secret_access_key` here as well?

**Comment by mercedesbh** [comment]
Looks great! Just one small question

File: internal/provider/data_source_vault_cluster.go (hunk 1 above)
Comment (bosouza) [reply internal/provider/data_source_vault_cluster.go:219]: done, funny how the code for preventing the value from the state file from being updated with "redacted" preventing this issue from surfacing in my tests :laughing:

**Review by mercedesbh — APPROVED** [review]

File: internal/provider/resource_vault_cluster.go [hunk 2]
```diff
@@ -1031,6 +1078,30 @@ func flattenObservabilityConfig(config *vaultmodels.HashicorpCloudVault20201125O
...
 	}
+
+	if cloudwatch := config.Cloudwatch; cloudwatch != nil {
+		configMap["cloudwatch_access_key_id"] = cloudwatch.AccessKeyID
+		configMap["cloudwatch_region"] = cloudwatch.Region
+		// ensure we only set properties that are defined in metrics/audit-logs streaming
+		if propertyName == "metrics_config" {
+			// Namespace is only used for streaming metrics
+			configMap["cloudwatch_namespace"] = cloudwatch.Namespace
+		} else {
```
Comment (jaireddjawed) [inline internal/provider/resource_vault_cluster.go:1089]: I noticed that a field called `dataset` is defined on the `config.ElasticSearch` struct. Would that be initialized here as well? Is it only defined when streaming metrics similar to the `cloudwatch_namespace` from earlier?

File: internal/provider/resource_vault_cluster.go [hunk 3]
```diff
@@ -1031,6 +1078,30 @@ func flattenObservabilityConfig(config *vaultmodels.HashicorpCloudVault20201125O
 			}
 		}
 	}
+
+	if cloudwatch := config.Cloudwatch; cloudwatch != nil {
+		configMap["cloudwatch_access_key_id"] = cloudwatch.AccessKeyID
+		configMap["cloudwatch_region"] = cloudwatch.Region
+		// ensure we only set properties that are defined in metrics/audit-logs streaming
+		if propertyName == "metrics_config" {
```
[the remark this answers is not in the document]
Comment (bosouza) [reply internal/provider/resource_vault_cluster.go:1086]: sorry, it's kinda messy. Have a look at `applyObservabilityConfigMutations` on `cloud-vault-service` to see what fields are automatically set by us when creating/updating the observability configs, not how they're different for metrics/logs. There's also `elsasticsearch.go` with the Vector definition for each of the providers.

[the remark this answers is not in the document]
Comment (bosouza) [reply internal/provider/resource_vault_cluster.go:1086]: let me know if it's still unclear and I can walk you through the relevant code.

Also, you could open the UI and see what fields are shown for logs/ metrics when configuring elasticsearch

**Pull request merged**